agent/command/test_selection_get.go

```
@@ -0,0 +1,113 @@
+	}
+
+	// Re-validate the command here, in case an expansion is not defined.
+	if err := c.validate(); err != nil {
+		return errors.WithStack(err)
+	}
+
+	// Resolve the output file path early so it's available for writing empty results
+	if !filepath.IsAbs(c.OutputFile) {
+		c.OutputFile = GetWorkingDirectory(conf, c.OutputFile)
+	}
+
+	if !c.isTestSelectionAllowed(conf) {
+		logger.Execution().Info("Test selection is not allowed/enabled, writing empty test list")
+		return c.writeTestList([]string{})
+	}
+
+	// Build the request using task information from TaskConfig.
+	request := model.SelectTestsRequest{
+		Project:      conf.Task.Project,
+		Requester:    conf.Task.Requester,
+		BuildVariant: conf.Task.BuildVariant,
+		TaskID:       conf.Task.Id,
+		TaskName:     conf.Task.DisplayName,
+		Tests:        c.Tests,
+	}
+
+	td := client.TaskData{ID: conf.Task.Id, Secret: conf.Task.Secret}
+	selectedTests, err := comm.SelectTests(ctx, td, request)
+	if err != nil {
+		return errors.Wrap(err, "calling test selection API")
+	}
+
+	// Write the results to the output file.
+	return c.writeTestList(selectedTests)
+}
+
+// isTestSelectionAllowed checks if test selection is allowed in the project and the running task
+func (c *testSelectionGet) isTestSelectionAllowed(conf *internal.TaskConfig) bool {
+	return utility.FromBoolPtr(conf.ProjectRef.TestSelection.Allowed) && conf.Task.TestSelectionEnabled
+}
+
+// writeTestList writes the list of tests to the output file as JSON in the required format
+func (c *testSelectionGet) writeTestList(tests []string) error {
+	testObjects := make([]map[string]string, len(tests))
+	for i, testName := range tests {
+		testObjects[i] = map[string]string{"name": testName}
+	}
+
+	output := TestSelectionOutput{
+		Tests: testObjects,
+	}
+
+	err := utility.WriteJSONFile(c.OutputFile, output)
+	if err != nil {
+		return errors.Wrap(err, "writing test selection output to file")
+	}
+	return nil
```

small nit: we can just return the errors.wrap since it's the last block in the function and errors.wrap returns nil for nil errors